source/core/parameters.cc

```diff
@@ -813,6 +813,114 @@ namespace Parameters
+      prm.declare_entry("penentration depth",
+                        "0.0",
+                        Patterns::Double(),
+                        "Penentration depth");
+      prm.declare_entry("beam radius",
+                        "0.0",
+                        Patterns::Double(),
+                        "Laser beam radius");
+
+
+      prm.enter_subsection("path");
+      laser_path = std::make_shared<Functions::ParsedFunction<dim>>(dim);
+      laser_path->declare_parameters(prm, dim);
+      if (dim == 2)
+        prm.set("Function expression", "0; 0");
+      if (dim == 3)
+        prm.set("Function expression", "0; 0; 0");
+      prm.leave_subsection();
+
+      prm.declare_entry("start time",
+                        "0.0",
+                        Patterns::Double(),
+                        "Start time of laser");
+      prm.declare_entry("end time",
+                        "0.0",
+                        Patterns::Double(),
+                        "End time of laser");
+
+      prm.declare_entry("beam direction",
+                        "z",
+                        Patterns::Selection("x|y|z"),
+                        "Laser beam direction "
+                        "Choices are <x|y|z>.");
+    }
+    prm.leave_subsection();
+  }
+
+  template <int dim>
+  void
+  Laser<dim>::parse_parameters(ParameterHandler &prm)
+  {
+    prm.enter_subsection("laser parameters");
+    {
+      activate_laser       = prm.get_bool("activate laser");
+      concentration_factor = prm.get_double("concentration factor");
+      laser_power          = prm.get_double("laser power");
+      laser_absorptivity   = prm.get_double("laser absorptivity");
+      penentration_depth   = prm.get_double("penentration depth");
+      beam_radius          = prm.get_double("beam radius");
+
+      prm.enter_subsection("path");
+      laser_path->parse_parameters(prm);
+      laser_path->set_time(0);
+      prm.leave_subsection();
+
+      start_time = prm.get_double("start time");
+      end_time   = prm.get_double("end time");
+
+      std::string op;
+      // Density
+      op = prm.get("beam direction");
+      if (op == "x")
+        {
+          beam_direction                     = BeamDirection::x;
+          beam_direction_coordinate          = 0;
+          perpendicular_plane_coordinate_one = 1;
+          perpendicular_plane_coordinate_two = 2;
+        }
+      else if (op == "y")
+        {
+          beam_direction                     = BeamDirection::y;
+          perpendicular_plane_coordinate_one = 0;
+          beam_direction_coordinate          = 1;
+          perpendicular_plane_coordinate_two = 2;
+        }
+      else if (op == "z")
+        {
+          beam_direction                     = BeamDirection::z;
+          perpendicular_plane_coordinate_one = 0;
+          perpendicular_plane_coordinate_two = 1;
+          beam_direction_coordinate          = 2;
+        }
+    }
+    prm.leave_subsection();
```

Comment:
I think I now understand what are the coordinates here. Maybe they should be called perpendicular_plane_direction_one and _two?

Reply:
As Bruno suggested to use coordinate instead of direction for the laser beam emission, I think it would be better to keep these names.